From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tabrisnet <tabris@tabris.net>
Date: Mon, Feb  9 02:36:30 2026 +0000
Subject: mt7988a bpi-r4 mt7988a-bananapi-bpi-r4-sd.dtb

OpenWRT uses some extra machinery for producing flattened DTBs that
aren't in the kernel. We don't support that, so instead this patches
the kernel tree Makefile.

Signed-off-by: tabrisnet <tabris@tabris.net>
---
 arch/arm64/boot/dts/mediatek/Makefile | 11 ++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/Makefile b/arch/arm64/boot/dts/mediatek/Makefile
index 652ed1c7f8c6..92776e05c0f9 100644
--- a/arch/arm64/boot/dts/mediatek/Makefile
+++ b/arch/arm64/boot/dts/mediatek/Makefile
@@ -119,12 +119,23 @@ dtb-$(CONFIG_ARCH_MEDIATEK) += mt8395-kontron-3-5-sbc-i1200.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8395-radxa-nio-12l.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8395-radxa-nio-12l-8-hd-panel.dtbo
 mt8395-radxa-nio-12l-8-hd-panel-dtbs := mt8395-radxa-nio-12l.dtb mt8395-radxa-nio-12l-8-hd-panel.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8395-radxa-nio-12l-8-hd-panel.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8516-pumpkin.dtb
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-emmc.dtb
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-sd.dtb
+
+mt7988a-bananapi-bpi-r4-sd-dtbs := \
+	mt7988a-bananapi-bpi-r4.dtb \
+	mt7988a-bananapi-bpi-r4-sd.dtbo
+mt7988a-bananapi-bpi-r4-emmc-dtbs := \
+	mt7988a-bananapi-bpi-r4.dtb \
+	mt7988a-bananapi-bpi-r4-emmc.dtbo
 
 # Device tree overlays support
+DTC_FLAGS_mt7988a-bananapi-bpi-r4-emmc := -@
+DTC_FLAGS_mt7988a-bananapi-bpi-r4-sd := -@
 DTC_FLAGS_mt7986a-bananapi-bpi-r3 := -@
 DTC_FLAGS_mt7986a-bananapi-bpi-r3-mini := -@
 DTC_FLAGS_mt7987a-bananapi-bpi-r4-lite := -@
 DTC_FLAGS_mt7988a-bananapi-bpi-r4 := -@
 DTC_FLAGS_mt7988a-bananapi-bpi-r4-2g5 := -@
-- 
Created with Armbian build tools https://github.com/armbian/build

